proto/pulumi/provider.proto

```diff
@@ -152,6 +152,13 @@ service ResourceProvider {
     // `Diff` compares an existing ("old") set of resource properties with a new set of properties and computes the
     // difference (if any) between them. `Diff` should only be called with values that have at some point been validated
     // by a [](pulumirpc.ResourceProvider.Check) call.
+    //
+    // The provider's response must satisfy the following invariants:
```

Comment:
Should this be on DiffResponse rather than here?

Reply:
Maybe? It just seems like it's easier overlooked there. This isn't really about peculiarities of the `DiffResponse` data structure but about the semantics of the `Diff` method.

Reply:
But don't those semantics also apply to DiffConfig?

Reply:
You tell me, I don't know DiffConfig or its internals!

Reply:
It hits the same paths as normal Diff so yes the same semantics should apply.